geoutils/raster/raster.py

```diff
@@ -398,6 +457,38 @@ def _check_cast_array_raster(
                 "Raster.from_array(array, array_transform, array_crs, array_nodata) then reproject."
             )
 
+    # If other is a single number
+    else:
+        other_data = other
+        nodata2 = None
+        dtype2 = rio.dtypes.get_minimum_dtype(other_data)
+
+    # Get raster dtype and nodata
+    nodata1 = raster.nodata
+    dtype1 = raster.data.dtype
+
+    # 1/ Output nodata depending on common data type
+    out_dtype = np.promote_types(dtype1, dtype2)
+
+    out_nodata = None
+    # Priority to nodata of first raster if both match the output dtype
+    if (nodata2 is not None) and (out_dtype == dtype2):
+        out_nodata = nodata2
+    elif (nodata1 is not None) and (out_dtype == dtype1):
+        out_nodata = nodata1
+    # For some cases the promoted output type is none if the inputs (e.g. inputs uint8 and int8 = int16 output)
```

Comment:
There is something missing in this line...

Reply:
The original text below was "For some cases the promote_types is neither (uint8 and int8 = int16),"

Reply:
Made clearer, might have to change with #517